src/prpl.ts

```diff
@@ -156,7 +164,15 @@ self.addEventListener('activate', () => self.registration.unregister());`);
       // We handle the caching header ourselves.
       cacheControl: false,
     };
-    send(request, fileToSend, sendOpts).pipe(response);
+
+    let stream = send(request, fileToSend, sendOpts);
+
+    // Set a custom error-handling function
+    if(config && config.error) {
+      stream = stream.on('error', config.error(request, response))
```

Comment:
There are two places above where an error can also be emitted (a `403` and a `500`). I think it would make sense to call the same custom error handler in those special cases too.

Reply:
Done.